### PR TITLE
[10.0][ADD][web_listview_range_select] Select a range of records with the shift key

### PR DESCRIPTION
**Pull request by tarteo**
Enables the user to select a range of records with the shift key.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Reading the description, better to call it `web_listview_range_select` :wink: 

Very good modules you are contributing anyway! I'll review it soon

**Comment by tarteo** [comment]
@pedrobaeza Good one, I'll change it! Thanks :).

**Review by rafaelbn — APPROVED** [review]
Tested in runbot

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
If you want to select several ranges, the module is not working correctly, as you can only do one range. The ideal is that the code go from the row where you have click with Shift pressed and find going up the first selected row (not the first absolute selected row). Do you understand me?

**Comment by tarteo** [comment]
@pedrobaeza Do you mean that the selected range does not reset if the user click + shift click multiple times?

like this:

Step 1:
normal click on row 0
shift click on row 3
selected rows are now 0 to 3

Step 2.1:
normal click on row 5
shift click on row 8
selected rows are now 5 to 8

Step 2.2:
shift click on row 5
selected rows are now 0 to 5

Currently it always extends the range resulting in rows 0 to 8 to be selected (if step 1 + step 2.1 are done).

**Comment by pedrobaeza** [comment]
That's the problem. What I want is to have selected multiple ranges. These are my expected steps:

Step 1:
normal click on row 0
shift click on row 3
selected rows are now 0 to 3

Step 2:
normal click on row 5
shift click on row 8
selected rows are now **0 to 3 and 5 to 8**

**Comment by tarteo** [comment]
Clear! I'll work on it ASAP.

**Comment by houssine78** [comment]
possible to cherry pick individual rows through CTRL + click?

**Comment by pedrobaeza** [comment]
Well, that's the normal behavior without Ctrl key

**Comment by houssine78** [comment]
ok I just get the thing(checkbox of the line is checked) I though that it was selecting all the lines for a copy paste(like you could do in excel).

**Comment by tarteo** [comment]
@houssine78 With Odoo's default behavior you already can cherry pick rows by just left clicking (without  using the ctrl key)

**Comment by tarteo** [comment]
@houssine78 Yes, this module only enables the user to select multiple rows with shift + click.

**Comment by houssine78** [comment]
ok thanks! it's already nice feature!

**Comment by tarteo** [comment]
@pedrobaeza It's now possible to select multiple ranges.

**Comment by pedrobaeza** [comment]
Cool, I'll check it when runbot finishes

File: web_listview_range_select/README.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+No configuration is needed.
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. click a record;
+#. hold shift and click another record.
```
Comment (pedrobaeza) [inline web_listview_range_select/README.rst]: You can add here:
```
#. You can repeat the operation as many times as you want.
```

**Review by pedrobaeza — APPROVED** [review]
Working flawlessly! Only one minor comment

File: web_listview_range_select/README.rst (hunk 1 above)
Comment (tarteo) [reply web_listview_range_select/README.rst]: Done :)

**Comment by espo-tony** [comment]
:+1: Great feature!

**Comment by legalsylvain** [comment]
Tested on runbot. works perfectly. Cool feature, thanks.
:+1: .

**Comment by pedrobaeza** [comment]
@legalsylvain please don't merge as is the PR, as you have included 7 useless commits, that are using more space, and are mixed on the commits view. Use "squash and merge" option if all commits come from the same author and can be grouped, or ask a logical squashing to the contributor otherwise.

**Comment by legalsylvain** [comment]
@pedrobaeza. Thanks for this information (squash and merge), i didn't know this new feature. I'll do it for the next PRs.

kind regards.

**Pull request merged**